bin/download-syntax

```diff
@@ -288,168 +294,241 @@ def _ts_to_js(scope: str, dct: Dict[str, Any]) -> Dict[str, Any]:
+        version_cmd = (
+            *git, 'log', '-1', '--format=%h', branch, '--', *repo.grammars,
+        )
+        version = subprocess.check_output(version_cmd).decode().strip()
 
-            version_cmd = (
-                *git, 'log', '-1', '--format=%h', branch, '--', *repo.grammars,
-            )
-            version = subprocess.check_output(version_cmd).decode().strip()
+        ancestor_cmd = (
+            *git, 'merge-base', '--is-ancestor', version, repo.version,
+        )
 
-            ancestor_cmd = (
-                *git, 'merge-base', '--is-ancestor', version, repo.version,
+        if (
+                repo.version == 'HEAD' or
+                version != repo.version and subprocess.call(ancestor_cmd)
+        ):
+            print(f'Updated {repo.name} {repo.version} => {version}')
+            return repo._replace(version=version)
+
+        print(f'{repo.name} is up to date!')
+        return repo
+
+
+def _update(*, only: Set[str]) -> int:
+    up_to_date_repos = []
+    with ThreadPoolExecutor() as executor:
+        futures = {}
+
+        for repo in REPOS:
+            if only and repo.name not in only:
+                up_to_date_repos.append(repo)
+                continue
+            futures[executor.submit(_update_repo, repo)] = repo
+
+        if not futures and only:
+            print(
+                'Nothing to do. Make sure you provided correct repo names to the "only" parameter',
+                file=stderr
             )
-            if (
-                    repo.version == 'HEAD' or
-                    version != repo.version and subprocess.call(ancestor_cmd)
-            ):
-                print(f'{repo.version} => {version}')
-                new.append(repo._replace(version=version))
+            return 1
+
+        for future in as_completed(futures):
+            try:
+                updated_repo = future.result()
+            except Exception as e:
+                print(f'error updating repo {futures[future]}', file=stderr)
+                continue
             else:
-                print('up to date!')
-                new.append(repo)
+                up_to_date_repos.append(updated_repo)
 
-    new.sort(key=lambda repo: repo.name)
+    up_to_date_repos.sort(key=lambda repo: repo.name)
 
-    if new != list(REPOS):
-        with open(__file__) as f:
+    if up_to_date_repos != list(REPOS):
+        print('Repositories have changed after the update, preparing to write changes...')
+        with open(__file__, encoding='utf-8') as f:
             contents = f.read()
 
         before, begin, rest = contents.partition('# BEGIN\n')
         _, end, rest = rest.partition('# END\n')
 
         new_contents = (
-            'REPOS = (\n' +
-            ''.join(f'{textwrap.indent(repr(r), " " * 4)},\n' for r in new) +
-            ')\n'
+                'REPOS = (\n' +
+                ''.join(f'{textwrap.indent(repr(r), " " * 4)},\n' for r in up_to_date_repos) +
+                ')\n'
```

Comment:
why did this change indent

Reply:
Uh, sorry, I probably messed up something in my IDE.